frontend/src/views/FuelCodes/AddFuelCode/AddEditFuelCode.jsx

```diff
@@ -0,0 +1,373 @@
+        alertRef.current?.triggerAlert({
+          message: 'Row deleted successfully.',
+          severity: 'success'
+        })
+      } catch (error) {
+        console.error('Error deleting row:', error)
+        alertRef.current?.triggerAlert({
+          message: `Error deleting row: ${error.message}`,
+          severity: 'error'
+        })
+      }
+    } else {
+      params.api.applyTransaction({ remove: [params.node.data] })
+    }
+    setModalData(null)
+  }
+
+  const handleApproveDraftCode = async (fuelCodeId) => {
+    await approveFuelCode(fuelCodeId)
+    await refetch()
+    setModalData(null)
+  }
+
+  const openDeleteModal = (fuelCodeId, params) => {
+    setModalData({
+      primaryButtonAction: () => handleDeleteFuelCode(fuelCodeId, params),
+      primaryButtonText: t('fuelCode:deleteFuelCodeBtn'),
+      secondaryButtonText: t('cancelBtn'),
+      title: t('fuelCode:deleteFuelCode'),
+      content: (
+        <Stack>
+          <BCTypography variant="h6">
+            {t('fuelCode:deleteFuelCode')}
+          </BCTypography>
+          <BCTypography mt={1} variant="body5">
+            {t('fuelCode:deleteConfirmText')}
+          </BCTypography>
+        </Stack>
+      )
+    })
+  }
+
+  const openApprovalModal = async (fuelCodeId) => {
+    setModalData({
+      primaryButtonAction: () => handleApproveDraftCode(fuelCodeId),
+      primaryButtonText: t('fuelCode:approveFuelCodeBtn'),
+      secondaryButtonText: t('cancelBtn'),
+      title: t('fuelCode:approveFuelCode'),
+      content: (
+        <Stack>
+          <BCTypography variant="h6">
+            {t('fuelCode:approveFuelCode')}
+          </BCTypography>
+          <BCTypography mt={1} variant="body5">
+            {t('fuelCode:approveConfirmText')}
+          </BCTypography>
+        </Stack>
+      )
+    })
+  }
+
+  const onCellEditingStopped = useCallback(
+    async (params) => {
+      if (params.oldValue === params.newValue) return
+
+      params.node.updateData({
+        ...params.node.data,
+        validationStatus: 'pending'
+      })
+
+      alertRef.current?.triggerAlert({
+        message: 'Updating row...',
+        severity: 'pending'
+      })
+
+      let updatedData = Object.entries(params.node.data)
+        .filter(([, value]) => value !== null && value !== '')
+        .reduce((acc, [key, value]) => {
+          acc[key] = value
+          return acc
+        }, {})
+
+      try {
+        setErrors({})
+        if (updatedData.fuelCodeId) {
+          await updateFuelCode(updatedData)
+        } else {
+          const res = await createFuelCode(updatedData)
+          updatedData.fuelCodeId = res.fuelCodeId
+        }
+
+        updatedData = {
+          ...updatedData,
+          validationStatus: 'success',
+          modified: false
+        }
+        alertRef.current?.triggerAlert({
+          message: 'Row updated successfully.',
+          severity: 'success'
+        })
+      } catch (error) {
```

Comment:
error handling could use a handler method which you could use throughout

something like:

```jsx

const handleError = (error, defaultMessage) => {
  console.error(error);
  const message =
    error.response?.data?.errors?.[0]?.message || defaultMessage;
  alertRef.current?.triggerAlert({
    message,
    severity: 'error',
  });
};

// Use in catch blocks
catch (error) {
  handleError(error, 'Error updating row');
}

Reply:
Good call, done! Cleaned up a few similar spots.